### DISCUSSION TITLE
Scrobbling to Jellyfin-server from offline mode?

### DISCUSSION THREAD

**Discussion by cyberwolfie1** [UnicornsOnLSD/finamp#926] [General]
I'm not sure how this stuff works, i.e. how playback data is reported to Jellyfin, but I was wondering if there was any way scrobbling of songs played offline could be realized? I would like this feature, and would want to submit an issue for it if it is possible, and if not, maybe submit an issue with Jellyfin to implement a way for clients to do this in the future.

**Comment by Chaphasilor** [comment]
The way scrobbling works is that the client tells the server what is currently being played, and updates that every time anything changes. Additionally, we also report that a track was started and that it was stopped/finished, so that plugins like "Playback Reporting" can record the duration.  
This is only possible to do when we can send data to the server, so not in offline mode. The Jellyfin server also sadly does not support submitting playback history later on.  

Now that I think about it, it would technically be possible to emulate offline playback when going back online, by pretending that Finamp is playing all of the tracks you played while in offline mode and effectively re-playing them in the background. This could probably even be done through a virtual "device", so that it shows in your Jellyfin devices/activity dashboard as "Finamp (Offline Listening)" or something.  
But it wouldn't be trivial to implement, and also pretty unreliable and use up battery in the background. The actual timestamps would of course also be wrong, since the server would just record the playback at the time Finamp reports it, and not at the time when it actually happened. So not a great solution overall.

Another option would be to import the playback history into just the "Playback Reporting" plugin's database. This would then still show up in your listening stats or Jellyfin Rewind, but Jellyfin itself wouldn't know about it (e.g. when sorting or reporting playback count).  
I'm planning to implement this for this year's Jellyfin Rewind.

However, Finamp (the beta version) already records offline playback history to a file, which you can export by going to the "Playback History" page through the sidebar and clicking the share button at the top right. This could then be used to manually import your listening activity into tools like last.fm, or even into the "Playback Reporting" plugin database.

So, if you want to see this in Jellyfin you should open a feature request for Jellyfin itself, because there's not a lot we can do about it in Finamp. You certainly have my vote!

**Comment by cyberwolfie1** [comment]
>  The Jellyfin server also sadly does not support submitting playback history later on.

Hm, seems like a good feature for the server to have. I could submit an issue on their repo (could not immediately find an existing repo for this). I'm sure any client that offers some sort of offline mode and scrobbling would want an easy way to push offline playback history back to the server on reconnect (with correct timestamps).

> Another option would be to import the playback history into just the "Playback Reporting" plugin's database. This would then still show up in your listening stats or Jellyfin Rewind, but Jellyfin itself wouldn't know about it (e.g. when sorting or reporting playback count).
I'm planning to implement this for this year's Jellyfin Rewind.

For my use case, I think this would be very good, even though I would love to retain the playback count.  Is this something that can be made automatic, or will it require some manual intervention to inject the offline playback history into the database?
